src/redux/ensRegistration.ts

```diff
@@ -0,0 +1,266 @@
+  } = getState();
+  const lcAccountAddress = accountAddress.toLowerCase();
+
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const registrationRecords = registration?.records || {};
+
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          records: { ...registrationRecords, [key]: value },
+        },
+      },
+    },
+  };
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_REGISTRATION_UPDATE_RECORDS,
+  });
+};
+
+export const removeRecordByKey = (
+  accountAddress: EthereumAddress,
+  key: string
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations, currentRegistrationName },
+  } = getState();
+
+  const lcAccountAddress = accountAddress.toLowerCase();
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const registrationRecords = registration?.records || {};
+
+  const newRecords = omit(registrationRecords, key) as Records;
+
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          records: newRecords,
+        },
+      },
+    },
+  };
+
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_REGISTRATION_REMOVE_RECORD_BY_KEY,
+  });
+};
+
+export const saveCommitRegistrationParameters = (
+  accountAddress: EthereumAddress,
+  registrationParameters: RegistrationParameters
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations, currentRegistrationName },
+  } = getState();
+
+  const lcAccountAddress = accountAddress.toLowerCase();
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          ...registrationParameters,
+        },
+      },
+    },
+  };
+
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_SAVE_COMMIT_REGISTRATION_PARAMETERS,
+  });
+};
+
+// -- Reducer ----------------------------------------- //
+const INITIAL_STATE: ENSRegistrationState = {
+  currentRegistrationName: '',
+  registrations: {},
+};
+
+export default (
+  state = INITIAL_STATE,
+  action: AnyAction
```

Comment:
TS can we define the actions more explicitly (can look at other redux TS files as examples)